packages/shared/src/components/auth/OnboardingRegistrationForm.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { ReactElement, useContext, useEffect, useState } from 'react';
+import { useMutation } from 'react-query';
+import { checkKratosEmail } from '../../lib/kratos';
+import { AuthFormProps, getFormEmail, providerMap } from './common';
+import OrDivider from './OrDivider';
+import AnalyticsContext from '../../contexts/AnalyticsContext';
+import { AuthEventNames, AuthTriggersOrString } from '../../lib/auth';
+import { ExperimentWinner } from '../../lib/featureValues';
+import { Button } from '../buttons/Button';
+import AuthForm from './AuthForm';
+import { TextField } from '../fields/TextField';
+import MailIcon from '../icons/Mail';
+import { IconSize } from '../Icon';
+import Alert, { AlertParagraph, AlertType } from '../widgets/Alert';
+
+const signupProviders = [providerMap.google, providerMap.github];
+
+interface OnboardingRegistrationFormProps extends AuthFormProps {
+  onExistingEmail?: (email: string) => unknown;
+  onSignup?: (email: string) => unknown;
+  onProviderClick?: (provider: string, login?: boolean) => unknown;
+  targetId?: string;
+  isLoginFlow?: boolean;
+  logInTitle?: string;
+  signUpTitle?: string;
+  trigger: AuthTriggersOrString;
+  isReady: boolean;
+}
+
+const OnboardingRegistrationForm = ({
+  onSignup,
+  onExistingEmail,
+  onProviderClick,
+  targetId,
+  isReady,
+  trigger,
+}: OnboardingRegistrationFormProps): ReactElement => {
+  const { trackEvent } = useContext(AnalyticsContext);
+  const [shouldLogin, setShouldLogin] = useState(false);
+  const [registerEmail, setRegisterEmail] = useState<string>(null);
+  const { mutateAsync: checkEmail, isLoading } = useMutation(
+    (emailParam: string) => checkKratosEmail(emailParam),
+  );
+
+  useEffect(() => {
+    trackEvent({
+      event_name: shouldLogin
+        ? AuthEventNames.OpenLogin
+        : AuthEventNames.OpenSignup,
+      extra: JSON.stringify({ trigger }),
+      target_id: targetId || ExperimentWinner.AuthVersion,
+    });
+    // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
should we just remove this rule entirely? I've rarely found it to be useful, since usually you know exactly what dependencies you need for a side effect or callback, and you rarely need to specify the full list